### ISSUE TITLE
Consider AVX instead of AVX2 when targetting Emscripten for WASM SIMD

### ISSUE THREAD

**Issue by briansemrau** [erincatto/box2c#116]
I found I was unable to compile box2c using Emscripten with AVX2 enabled. Currently AVX2 is not supported by Emscripten:
>Currently only the SSE1, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, and 128-bit AVX instruction sets are supported.
 (https://emscripten.org/docs/porting/simd.html#compiling-simd-code-targeting-x86-sse-instruction-sets)

Disabling AVX2 caused build errors:
```cmake
set(BOX2D_AVX2 OFF)
```
```
box2c/src/contact_solver.c:730:6: error: loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Werror,-Wpass-failed=transform-warning]
```
I worked around this by removing `set(CMAKE_COMPILE_WARNING_AS_ERROR ON)` in `box2c/CMakeLists.txt`. So as of commit 9d7f1ec you cannot build box2c with Emscripten out of the box.

To enable SIMD with Emscripten I opted to enable AVX instead of AVX2. I have no clue if this is really the best option for performance, nor have I confirmed whether anything properly works yet. But it does compile.

These are the changes I made:
```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 337aa14..151d164 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -166,6 +166,16 @@ elseif (MINGW)
        endif()
 elseif (APPLE)
        # target_compile_options(box2d PRIVATE)
+elseif (EMSCRIPTEN)
+       message(STATUS "Box2D using Emscripten")
+       if (BOX2D_AVX2)
+               message(STATUS "Box2D using AVX instead of AVX2")       
+               target_compile_options(box2d PRIVATE -mavx -msimd128)
+       else()
+               # see SIMDE_DIAGNOSTIC_DISABLE_PSABI_
+               message(STATUS "Box2D disabling ABI warning")
+               target_compile_options(box2d PRIVATE -Wno-psabi)
+       endif()
 elseif (UNIX)
        message(STATUS "Box2D using Unix")
        if ("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "aarch64")
```

**Comment by erincatto** [comment]
Oh yeah, the PSABI warnings. I've seen those. It is related to the SIMDE library. I asked about this here:
https://app.gitter.im/#/room/#simd-everywhere_community

> Hello! I'm build in C on gcc and using the avx2.h include but not enabling mavx2 (as an option for users that don't have AVX2). I'm getting this warning: "warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]". How should I deal with this? This is all internal code and not in my library API.

> Hello Erin. You can compile with -Wno-psabi or insert a SIMDE_DIAGNOSTIC_DISABLE_PSABI_ before the internal code that is causing this. If you use the later, don't forget to wrap that section in a HEDLEY_DIAGNOSTIC_PUSH/HEDLEY_DIAGNOSTIC_POP pair or equivalent

https://github.com/simd-everywhere/simde/blob/5405bbdcc7e9045b0901c847c8868594deb511a6/simde/simde-diagnostic.h#L82

**Comment by erincatto** [comment]
Fixed in #126
This setup should use SSE2 on Emscripten. AVX but not AVX2 is not worth it.
